src/piptool.py

```diff
@@ -4,15 +4,15 @@
 import sys
 
 from pip._internal.req.req_file import parse_requirements
-from pip._internal.download import PipSession
-
+from pip._internal.network.session import PipSession
+from pip._vendor.packaging.requirements import Requirement
 
 def clean_name(name):
     # Escape any illegal characters with underscore.
     return re.sub("[-.+]", "_", name)
 
 
-def is_pinned_requirement(ireq):
+def is_pinned_requirement(preq, ireq = None):
```

Comment:
seems like you just need `editable` from preq,  so just pass that, also plz rename ireq to req, it was supposed to be InstallRequirement, it would be great if you can add some type notation comments. also plz update docstrings